op-node-derivers.md

```diff
@@ -0,0 +1,573 @@
+
+### Interop
+
+With interop, we are adding an entirely new block safety type:
+["cross-unsafe"](https://github.com/ethereum-optimism/specs/blob/main/specs/interop/verifier.md#cross-unsafe-inputs).
+
+This requires the op-node to:
+- asynchronously validate "unsafe" blocks to become "cross-unsafe".
+- block "safe" progression until blocks are "cross-unsafe" first (applicable to L1-derived block attributes also).
+- maintain a view of external L2 chains, and reorg local L2 chain-state when remote L2 chain-state is invalidated.
+
+### Alt-DA
+
+To support alt-DA and other customizations,
+there is a desire to extend the driver and derivation behavior with a bundled interface.
+
+Adding or modifying functionality should not require many sparse changes.
+Rather, we want a more plugin-like interface, that allows something to fit in and override core functionality,
+while reusing default functionality where desired, and still being structured to be fault-provable.
+
+Extending the stack should not require duplication of the state-transition 
+modifications between the proof system and regular node.
+
+Alt-DA also changes the meaning of "inputs";
+DA commitments alone are not sufficient to finalize data as we would with rollup data,
+as the underlying data is not finalized when the L1-registration of the commitment is.
+With plasma, a challenge period has to expire first.
+In other words, alt-DA changes the finality system.
+See [Alt-DA finalization specs](https://github.com/ethereum-optimism/design-docs/blob/8d66ce7bd1ec09f9d4f6ebb7765844b03440ed36/plasma-mode.md#finalization)
+for additional details.
+
+# Alternatives Considered
+
+<!-- *List out a short summary of each possible solution that was considered.
+Comparing the effort of each solution* --> 
+
+### Status-quo
+
+To integrate interop in the op-node as-is, we would have to:
+- Modify the L2 forkchoice state to include cross-unsafe.
+- Modify the payload-consolidation to have a cross-unsafe pre-requisite.
+- Modify the payload-force-processing to not instantly produce a safe block (that would bypass cross-unsafe).
+- Modify the various forkchoice-updates of EngineController to be consistent with the cross-unsafe head.
+- Modify the engine-queue/controller to allow reorgs through a different path than the current block-attributes path.
+- Modify the find-sync-start loop to add a block tag of cross-unsafe.
+- Modify the driver to accomodate for a special-purpose process that maintains a view of the cross-L2 safety,
+  while explicitly synchronizing the engine-control access in the main state-loop.
+
+This is high-effort (medium in size, but hard to test), and accumulates a lot more tech-debt,
+since it makes everything less extensible and adds complexity without simplifying any existing code.
+
+### Many `RepeatCond` and some shared locks
+
+See design doc [here (private)](https://github.com/ethereum-optimism/protocol-quest/issues/202).
+The gist is that it centers around conditions/effects,
```

Comment:
Is "it" referring to the "existing rollup driver implementation"? Might be good to use the more descriptive term here

Reply:
Nvm, I see "it" is referring to the proposed alternative design described in detail within the linked issue